src/wrapper/objects/auto_byte_array.rs

```diff
@@ -58,17 +58,45 @@ impl<'a, 'b> AutoByteArray<'a, 'b> {
         self.ptr.as_ptr()
     }
 
-    /// Commits the result of the array, if it is a copy
+    fn commit_byte_array_elements(&mut self) -> Result<()> {
+        jni_void_call!(
+            self.env.get_native_interface(),
+            ReleaseByteArrayElements,
+            *self.obj,
+            self.ptr.as_mut(),
+            sys::JNI_COMMIT
+        );
+        Ok(())
+    }
+
+    fn release_byte_array_elements(&mut self) -> Result<()> {
```

Comment:
Here and in the critical flavour: would it make sense to add an utility method 
```
fn release_byte_array_elements(&mut self, i32)
```

and then reusing it in both of these methods:
```
commit: release_byte_array_elements(sys::JNI_COMMIT)

release: release_byte_array_elements(mode as i32)
```

Reply:
This could be an option; particularly now that these methods are private. The problem is with the naming, that comes from the JNI overloading of release and commit... I would leave it as it is, just for clarity.

Reply:
On second though, I think you're right. The new version is more compact, and the naming just follows the original JNI impl.

Reply:
Not sure I get the naming issue: if a new method is named `release_array_elements` and accepts a release mode, it will be the same as the JNI method, but with some of its arguments already applied (= coming from the struct fields). We can then inline the other two private methods (I see commit is already inlined), or rename them to "release"/"commit".

Reply:
Yep. See the latest version.